src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIProxy.cs

```diff
@@ -238,28 +243,190 @@ private SNITCPHandle CreateTcpHandle(string fullServerName, long timerExpire, ob
+                hostIPAddress = addressList.First(a => a.AddressFamily == family);
+                if (hostIPAddress == null)
+                {
+                    return false;
+                }
+            }
+            catch
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Sends CLNT_UCAST_INST request for given instance name to SQL Sever Browser, and receive SVR_RESP from the Browser.
+        /// </summary>
+        /// <param name="browserEP">SQL Sever Browser IPEndPoint</param>
+        /// <param name="instanceName">instance name for CLNT_UCAST_INST request</param>
+        /// <param name="responsePacket">SVR_RESP packets from SQL Sever Browser</param>
+        /// <returns>bool value showing if operation is successful</returns>
+        private static bool Send_CLNT_UCAST_INST(IPEndPoint browserEP, string instanceName, out byte[] responsePacket)
+        {
+            responsePacket = null;
+
+            if (browserEP == null || String.IsNullOrWhiteSpace(instanceName))
+            {
+                return false;
+            }
+
+            List<byte> byteList = new List<byte>(new byte[] { 0x04 });
+            byteList.AddRange(Encoding.ASCII.GetBytes(instanceName));
+            byte[] requestPacket = byteList.ToArray();
+
+            bool success = true;
+            UdpClient client = null;
+            try
+            {
+                client = new UdpClient(AddressFamily.InterNetwork);
+                Task<int> sendTask = client.SendAsync(requestPacket, requestPacket.Length, browserEP);
+                Task<UdpReceiveResult> receiveTask = null;
+                if (sendTask.Wait(1000) && (receiveTask = client.ReceiveAsync()).Wait(1000))
+                {
+
+                    responsePacket = receiveTask.Result.Buffer;
+                }
+                else
+                {
+                    success = false;
+                }
+            }
+            catch
+            {
+                success = false;
+            }
+            finally
+            {
+                if (client != null)
+                {
+                    client.Dispose();
+                }
+            }
+
+            return success;
+        }
+
+        /// <summary>
+        /// Finds port number for given instance name.
+        /// </summary>
+        /// <param name="browserHostname">SQL Sever Browser hostname</param>
+        /// <param name="instanceName">instance name to find port number</param>
+        /// <param name="instancePort">port number for given instance name</param>
+        /// <param name="browserPort">>SQL Sever Browser port. Default value is 1434</param>
+        /// <returns>bool value showing if operation is successful</returns>
+        private static bool GetPortByInstanceName(string browserHostname, string instanceName, out int instancePort, int browserPort = 1434)
```

Comment:
Is the `browserPort` parameter needed for the method? Is the Sql browser process expected to not run on the Port 1434? 
If it always runs on 1434, I recommend changing this to a class constant.